R/utils.R

```diff
@@ -99,12 +99,31 @@ is_quantity <- function(x) {
   x
 }
 
+vctrs_list_compat <- function(x, error_call = caller_env(), error_arg = caller_arg(x)) {
+  force(error_arg)
```

Comment:
Is this required? I thought part of the point of passing `error_call` and `error_arg` through in the specific way that we do is to avoid forcing the promises until they are actually required

Reply:
I had to do it to get a good error.

Reply:
Maybe preserve `x` by assigning the result of `vctrs_vec_compat(x)` to `out` instead of `x`?

I agree that the default pattern should be to evaluate these things lazily.

Reply:
Oh right, I couldn't figure out what the problem was but now I see that what Lionel suggests is a nice simple fix that maintains the standard pattern

```r
vctrs_list_compat <- function(x, error_call = caller_env(), error_arg = caller_arg(x)) {
  out <- vctrs_vec_compat(x)
  vec_check_list(out, call = error_call, arg = error_arg)
  out
}
```

Reply:
Oh yeah, that's better. I pushed a fix to main.